File: Phase-2/frontend/types/task.ts
export interface Task {
  id: string;
  title: string;
  description?: string;
  status: 'active' | 'completed';
  priority: 'low' | 'medium' | 'high';
  due_date?: string; // ISO string
  completed_at?: string | null; // ISO string
  created_at: string; // ISO string
  updated_at: string; // ISO string
  user_id: string;
}

export interface TaskCreateData {
  title: string;
  description?: string;
  status?: 'active' | 'completed';
  priority?: 'low' | 'medium' | 'high';
  due_date?: string; // ISO string
}

export interface TaskUpdateData {
  title?: string;
  description?: string;
  status?: 'active' | 'completed';
  priority?: 'low' | 'medium' | 'high';
  due_date?: string; // ISO string
}

File: Phase-2/backend/app/models/task.py
from sqlmodel import SQLModel, Field, Relationship
from datetime import datetime
from uuid import UUID, uuid4
from typing import Optional
from app.models.user import User
import enum


class TaskStatus(str, enum.Enum):
    ACTIVE = "active"
    COMPLETED = "completed"


class TaskPriority(str, enum.Enum):
    LOW = "low"
    MEDIUM = "medium"
    HIGH = "high"


class TaskBase(SQLModel):
    title: str = Field(min_length=1, max_length=255)
    description: Optional[str] = Field(default=None, max_length=1000)
    status: TaskStatus = Field(default=TaskStatus.ACTIVE)
    priority: TaskPriority = Field(default=TaskPriority.MEDIUM)
    due_date: Optional[datetime] = Field(default=None)
    user_id: UUID = Field(foreign_key="users.id", nullable=False)


class Task(TaskBase, table=True):
    __tablename__ = "tasks"

    id: UUID = Field(default_factory=uuid4, primary_key=True)
    completed_at: Optional[datetime] = Field(default=None)
    created_at: datetime = Field(default_factory=datetime.utcnow, nullable=False)
    updated_at: datetime = Field(default_factory=datetime.utcnow, nullable=False)

    # Relationship to User
    user: User = Relationship(back_populates="tasks")

    class Config:
        arbitrary_types_allowed = True